File: Core/src/main/java/net/winternetwork/bedwars/core/commands/GeneratorCommands.java
package net.winternetwork.bedwars.core.commands;

import me.saiintbrisson.commands.Execution;
import me.saiintbrisson.commands.annotations.Command;
import net.winternetwork.bedwars.core.Core;
import net.winternetwork.bedwars.core.game.generators.Generator;
import net.winternetwork.bedwars.core.game.generators.GeneratorManager;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

public class GeneratorCommands {

    private final GeneratorManager manager = GeneratorManager.getInstance();

    @Command(
            name = "generators",
            permission = "generators.use"
    )
    public void sendHelp(Execution execution) {
        execution.sendMessage(new String[]{
                "",
                "/generators create <nome> - segurando o item na mão.",
                "/generators setBlock <nome> - e depois clica no bloco",
                ""
        });
    }

    @Command(
            name = "generators.create",
            permission = "generators.use",
            usage = "generators create <nome>",
            inGameOnly = true
    )
    public void create(Execution execution, String name) {
        ItemStack hand = execution.getPlayer().getItemInHand();

        if (hand == null || hand.getType() == Material.AIR) {
            execution.sendMessage("Você não está segurando nenhum item na mão!");
            return;
        }

        Generator generator = new Generator(name, hand.getType());

        manager.put(generator);

        execution.sendMessage(
                generator.toString()
        );
    }

    @Command(
            name = "generators.setblock",
            permission = "generators.use",
            usage = "generators setblock <nome>",
            inGameOnly = true
    )
    public void setBlock(Execution execution, String name) {
        Generator generator = manager.get(name);

        if (generator == null) {
            execution.sendMessage("não foi encontrado nenhum gerador com esse nome.");
            return;
        }

        Core.getInstance().getGenBlockMap().put(execution.getPlayer().getName(), name);

        execution.sendMessage("§eClique em um bloco.");
    }

}

File: Game/src/main/kotlin/net/winternetwork/bedwars/modules/generators/command/GeneratorCommands.kt
package net.winternetwork.bedwars.modules.generators.command

import me.saiintbrisson.minecraft.command.annotation.Command
import me.saiintbrisson.minecraft.command.command.Context
import me.saiintbrisson.minecraft.command.target.CommandTarget
import net.winternetwork.bedwars.modules.generators.manager.GeneratorManager
import net.winternetwork.bedwars.modules.generators.model.Generator
import org.bukkit.Material
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class GeneratorCommands(
        private val generatorManager: GeneratorManager,
        private val setupMap: MutableMap<String, String>
) {

    @Command(
            name = "generators",
            permission = "generators.use",
            target = CommandTarget.PLAYER
    )
    fun help(ctx: Context<CommandSender>) {
        ctx.sendMessage(
                arrayOf(
                        "",
                        "§e§l/generators §fcreate §e<nome> §6- segurando item na mão.",
                        "§e§l/generators §fsetblock §e<nome> §6- e depois clica no bloco",
                        ""
                )
        )
    }

    @Command(
            name = "generators.create",
            permission = "generators.use",
            usage = "generators create <nome>"
    )
    fun create(ctx: Context<Player>, name: String) {
        val hand = ctx.sender.itemInHand
        if (hand == null || hand.type == Material.AIR) {
            ctx.sendMessage("§cVocê não está segurando nenhum item na mão.")
            return
        }

        Generator(name, hand.type).run {
            generatorManager.add(this.name, this)
            ctx.sendMessage(this.toString())
        }
    }

    @Command(
            name = "generators.setblock",
            permission = "generators.use",
            usage = "generators setblock <nome>"
    )
    fun setBlock(ctx: Context<Player>, name: String) {
        val generator = generatorManager[name]
        if (generator == null) {
            ctx.sendMessage("não foi encontrado nenhum gerador com esse nome.")
            return
        }

        setupMap[ctx.sender.name] = name
        ctx.sendMessage("§aClique em um bloco.")
    }
}